src/tools/adig.c

```diff
@@ -138,14 +145,77 @@ static void print_help(void)
 }
 
 static ares_bool_t read_cmdline(int argc, const char * const *argv,
-                                adig_config_t *config)
+                                adig_config_t *config, ares_bool_t is_rcfile)
 {
-  ares_getopt_state_t state;
-  int                 c;
+  ares_getopt_state_t  state;
+  int                  c;
+  char                 configdir[256];
+  char                *configdir_xdg;
+  unsigned int         cdlen = 0;
+  unsigned int         cdsize = sizeof(configdir);
+  char                *homedir;
 
   ares_getopt_init(&state, argc, argv);
   state.opterr = 0;
 
+  if (is_rcfile) {
+    state.optind = 0;
+  } else {
```

Comment:
This whole code block should be pulled out into a helper function that is called from within the main() function.  So you'll process the RCfile first then the command line args.

Reply:
Done!